Source/WebCore/page/EventHandler.cpp

```diff
@@ -3277,8 +3277,10 @@ bool EventHandler::sendContextMenuEvent(const PlatformMouseEvent& event)
     if (mouseEvent.scrollbar() || view->scrollbarAtPoint(event.position()))
         return false;
 
+    auto* page = m_frame.page();
     if (m_frame.editor().behavior().shouldSelectOnContextualMenuClick()
-        && !m_frame.selection().contains(viewportPos)) {
+        && !m_frame.selection().contains(viewportPos)
+        && page && (!page->isEditable() || !is<HTMLImageElement>(mouseEvent.hitTestResult().targetElement()))) {
```

Comment:
So this will happen only when API clients enable editing?
How do we know this new behavior is compatible with existing apps?

Reply:
The only existing apps that use the SPI are Notes and Mail, and I think this behavior is desirable in both of those apps, since it also aligns with native `NSTextView` behavior.

Is the concern about third party apps that use WebKit on macOS, that also use the `_editable` SPI?

Reply:
Right. Given we don't have this special behavior on the Web, it's also possible that some apps are relying on the existence behavior. I guess we'd only find out via landing this and testing third party apps.

Reply:
Yeah, there's definitely a risk of breaking WebKit-based third party apps that use this SPI — however, since this is SPI to begin with (as opposed to public API), I think it's okay to adjust our behaviors here to better match the platform as long as it suits the needs of all of our internal clients.

I think that to err on the side of caution, we could additionally put this behind a linked-on-or-after check to avoid any bincompat fallout that may occur with third party apps that happen to use this SPI. @kcheney1 (and @hortont424, and @pxlcoder), what do you all think!

Reply:
I mentioned this in Slack, but I think the fix is niche enough that it seems low risk. Open to opinions here because I am not an expert though.

Reply:
It is niche, but the following scenario could be affected:

1. `preventDefault` on the `contextmenu` event.
2. Showing a custom menu.
3. One of the actions operates on the selection.

A linked-on-or-after check would be safer, but I'm also not sure we normally do that for SPI?

Reply:
FWIW, I think it's okay to land as is if we notify QA people that this change is happening but SDK version check is certainly safer (although it might be an overkill in the absence of any brokerage).

Reply:
> A linked-on-or-after check would be safer, but I'm also not sure we normally do that for SPI?

Indeed, we don't really introduce LoA for behavior changes to SPI, but it is true that on macOS, there are non-app store apps that use WebKit SPI in ways that are similar to how internal apps would. IMO it's _technically feasible_ (albeit, fairly unlikely) that such an app might be depending on this behavior.

This does make me wonder whether we should scour GitHub for any frameworks or apps that might call into `-[WKWebView _setEditable:YES]`, which might be affected...